validator/validators/basic/basic_validator.go

```diff
@@ -49,6 +49,8 @@ func (s *BasicValidator) GenerateLoad() error {
 	switch dataType {
 	case "logs":
 		return common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, metricSendingInterval, dataRate)
+	case "traces":
+		return util.StartTraceGeneration(receiver,agentConfigFilePath,agentCollectionPeriod,metricSendingInterval)
```

Comment:
nit: It looks like the log generation is in common. Consider moving this there as well for consistency.